processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -108,4 +110,22 @@ else if(o instanceof Number)
     }
     throw new IAE("Cannot convert [%s] to QueryGranularity", o.getClass());
   }
+
+  //simple merge strategy on query granularity that checks if all are equal or else
+  //returns null. this can be improved in future but is good enough for most use-cases.
+  public static QueryGranularity mergeQueryGranularities(List<QueryGranularity> toMerge)
+  {
+    if (toMerge == null || toMerge.size() == 0) {
+      return null;
+    }
+
+    QueryGranularity result = toMerge.get(0);
+    for (int i = 1; i < toMerge.size(); i++) {
+      if (!Objects.equals(result, toMerge.get(i))) {
+        return null;
```

Comment:
@nishantmonu51 merging query granularities is a matter under discussion, see  https://github.com/druid-io/druid/issues/1514 . i will probably make the merge smarter in a separate PR. current simple strategy works fine for the use case when you are looking at metadata of individual segments or if granularity stayed same for the segments (which is true most of the time).